### PR TITLE
Implement a common utility for retrieving crm nexthop counter.

### PR DESCRIPTION
**Pull request by bingwang-ms**
Signed-off-by: bingwang <bingwang@microsoft.com>

### Description of PR

Summary:
Fixes # (issue)

1. Add a autoused module level fixture to set and recover crm polling interval
2. Add a common utility to retrieve crm nexthop counter.

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
This PR is to implement a common utility for retrieving crm nexthop counter.

#### How did you do it?
Please refer to code.

#### How did you verify/test it?
Verified with a demo script.
```
def test_counter(lower_tor_host):
    counter = crm_nexthop_counter(lower_tor_host)
    logging.info("counter = %d" % counter)

```
#### Any platform specific information?
No.

#### Supported testbed topology if it's a new test case?
duator testbed specific.

### Documentation 

No.

### CODE REVIEW

File: tests/dualtor/conftest.py [hunk 1]
```diff
@@ -1,2 +1,23 @@
...
+import logging
+import time
+from tests.common.dualtor.mux_simulator_control import mux_server_url, toggle_all_simulator_ports_to_upper_tor # lgtm[py/unused-import]
+from tests.common.dualtor.dual_tor_utils import lower_tor_host, upper_tor_host # lgtm[py/unused-import]
+
+CRM_POLL_INTERVAL = 1
+CRM_DEFAULT_POLL_INTERVAL = 300
+
+@pytest.fixture(scope='module', autouse=True)
+def set_crm_polling_interval(lower_tor_host):
```
Comment (neethajohn) [inline tests/dualtor/conftest.py:11]: This doesn't look like the right place to put this fixture if it is only needed for crm test

File: tests/dualtor/conftest.py [hunk 2]
```diff
@@ -1,2 +1,23 @@
...
+    """
+    A session level fixture to set crm polling interval to 1 second
```
Comment (neethajohn) [inline tests/dualtor/conftest.py:13]: A module level fixture

File: tests/dualtor/conftest.py (hunk 1 above)
Comment (bingwang-ms) [reply tests/dualtor/conftest.py:11]: Thanks. I change the fixture to function level, and not autouse.

File: tests/dualtor/conftest.py (hunk 2 above)
Comment (bingwang-ms) [reply tests/dualtor/conftest.py]: Thanks. Updated.

**Review by neethajohn — APPROVED** [review]

File: tests/dualtor/conftest.py [hunk 3]
```diff
@@ -1,2 +1,23 @@
...
+    A function level fixture to set crm polling interval to 1 second
+    """
+    wait_time = 2
+    lower_tor_host.command("crm config polling interval {}".format(CRM_POLL_INTERVAL))
+    logging.info("Waiting {} sec for CRM counters to become updated".format(wait_time))
+    time.sleep(wait_time)
+    yield
+    lower_tor_host.command("crm config polling interval {}".format(CRM_DEFAULT_POLL_INTERVAL))
+    logging.info("Waiting {} sec for CRM counters to become updated".format(wait_time))
+    time.sleep(wait_time)
```
Comment (yxieca) [inline tests/dualtor/conftest.py:22]: why do you need to wait on the restore part?

File: tests/common/dualtor/dual_tor_utils.py [hunk 4]
```diff
@@ -557,3 +557,13 @@ def check_tunnel_balance(ptfhost, active_tor_mac, standby_tor_mac, active_tor_ip
                 log_file=log_file,
                 qlen=2000,
                 socket_recv_size=16384)
+
+
+def crm_nexthop_counter(host):
```
Comment (yxieca) [inline tests/common/dualtor/dual_tor_utils.py:562]: suggest name this function as get_crm_nexthop_counter

File: tests/dualtor/conftest.py (hunk 3 above)
Comment (bingwang-ms) [reply tests/dualtor/conftest.py]: Not necessary, already removed. Thanks.

File: tests/common/dualtor/dual_tor_utils.py (hunk 4 above)
Comment (bingwang-ms) [reply tests/common/dualtor/dual_tor_utils.py]: Updated. Thanks!

**Comment by yxieca** [comment]
@bingwang-ms conftest.py has some unused import with comments stating so. What was the reason of having them there? Should I not remove them?

File: tests/common/dualtor/dual_tor_utils.py [hunk 5]
```diff
@@ -557,3 +557,12 @@ def check_tunnel_balance(ptfhost, active_tor_mac, standby_tor_mac, active_tor_ip
...
                socket_recv_size=16384)
+
+
+def get_crm_nexthop_counter(host):
+    """
+    Get used crm nexthop counter
+    """
+    cmd = "redis-cli --raw -n 2 HMGET CRM:STATS crm_stats_ipv4_nexthop_used"
+    output = host.shell(cmd)['stdout']
+    return int(output)
```
Comment (lolyu) [inline tests/common/dualtor/dual_tor_utils.py:568]: Need an extra blank line here.

Comment (bingwang-ms) [reply tests/common/dualtor/dual_tor_utils.py]: Fixed. Thanks!

**Pull request merged**